include/common/core/allocator.hpp

```diff
@@ -0,0 +1,153 @@
+#include <algorithm>
+#include <array>
+#include <variant>
+
+#include "common/core/logging.h"
+
+/*
+ * A generic custom allocator class that can be used to statically allocate
```

Comment:
Might want to fix up this comment, looks like the formatter chewed it up pretty bad